[Android] custom release/version/dist is set to null after event is sent

OS:
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

_Platform:_
- [ ]  iOS
- [x]  Android

Output of `node -v && npm -v && npm ls --prod --depth=0`

```
v9.3.0
5.6.0
pzrb2b@1.8.1 /Users/honza/Projects/pzrb2b
├── chroma-js@1.3.5
├── content-disposition@0.5.2
├── diacritics@1.3.0
├── hoist-non-react-statics@2.3.1
├── humps@2.0.1
├── intl@1.2.5
├── invariant@2.2.2
├── mime-types@2.1.17
├── moment@2.20.1
├── normalizr@3.2.4
├── prop-types@15.6.0
├── query-string@5.0.1
├── ramda@0.25.0
├── react@16.2.0
├── react-intl@2.4.0
├── react-native@0.52.0
├── react-native-branch@2.2.3
├── react-native-code-push@5.2.1
├── react-native-config@0.11.4
├── react-native-device-info@0.13.0
├── react-native-fbsdk@0.6.3
├── react-native-fcm@11.2.0
├── react-native-fetch-blob@0.10.8
├── react-native-google-analytics-bridge@5.4.1
├── react-native-keyboard-aware-scroll-view@0.4.1
├── react-native-mixpanel@0.0.16 invalid
├── react-native-modal-dropdown@0.5.0
├── react-native-orientation@3.1.0
├── react-native-sentry@0.32.0
├── react-native-share@1.0.23
├── react-native-star-rating@1.0.8
├── react-native-svg@5.5.1
├── react-native-swiper@1.5.13
├── react-native-vector-icons@4.5.0
├── react-native-version-check@2.2.0
├── react-navigation@1.0.0-beta.23
├── react-redux@5.0.6
├── recompose@0.26.0
├── redux@3.7.2
├── redux-form@7.2.0
├── redux-logger@3.0.6
├── redux-observable@0.17.0
├── redux-persist@4.9.1
├── reselect@3.0.1
├── rxjs@5.5.6
└── validator@9.2.0

npm ERR! invalid: react-native-mixpanel@0.0.16 /Users/honza/Projects/pzrb2b/node_modules/react-native-mixpanel
```

Config:

```
Sentry.config('https://...@sentry.io/...').install()
```

---
I have following issue:

On android (ios is fine) after first event is sent the custom values for release/version/dist are set null so next subsequent events have release and dist values null.

This is caused by function _stripInternalSentry_ defined _RNSentryModule.java:415_ which clears version variables each time event is sent. This behaviour is inconsistent with ios, which retains those values for the whole time.

**Steps to reproduce:**
- `Sentry.setVersion('1.1.0-foobar')`
- `Sentry.captureException(new Error('First error'))`
- `Sentry.captureException(new Error('Second error'))`

**Actual result:**
On sentry.io I see two events with releases
- 'First error' with release _org.sample.pzrb2b-1.1.0-foobar_
- 'Second error' with release _org.sample.pzrb2b-null_

**Expected result:**
- 'First error' with release _org.sample.pzrb2b-1.1.0-foobar_
- 'Second error' with release _org.sample.pzrb2b-1.1.0-foobar_

--- a/android/src/main/java/io/sentry/RNSentryModule.java
+++ b/android/src/main/java/io/sentry/RNSentryModule.java
@@ -295,7 +295,6 @@ public class RNSentryModule extends ReactContextBaseJavaModule {
         androidHelper.helpBuildingEvent(eventBuilder);
 
         setRelease(eventBuilder);
-        stripInternalSentry(eventBuilder);
         eventBuilder.withBreadcrumbs(Sentry.getStoredClient().getContext().getBreadcrumbs());
 
         if (extra != null) {
@@ -410,16 +409,6 @@ public class RNSentryModule extends ReactContextBaseJavaModule {
             }
         }
         return "";
-    }
-
-    private static void stripInternalSentry(EventBuilder eventBuilder) {
-        if (extra != null) {
-            for (Map.Entry<String, Object> entry : extra.toHashMap().entrySet()) {
-                if (entry.getKey().startsWith("__sentry")) {
-                    extra.putNull(entry.getKey());
-                }
-            }
-        }
     }
 
     private static void setRelease(EventBuilder eventBuilder) {

--- a/ios/RNSentry.m
+++ b/ios/RNSentry.m
@@ -16,8 +16,6 @@ NSString *const RNSentrySdkName = @"sentry-react-native";
 
 @property (nonatomic, strong) NSMutableDictionary *moduleMapping;
 
-@property (nonatomic, strong) SentryClient *client;
-
 @end
 
 
@@ -121,8 +119,7 @@ RCT_EXPORT_MODULE()
 RCT_EXPORT_METHOD(crashedLastLaunch:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject) {
     NSNumber *crashedLastLaunch = @NO;
-
-    if ([self.client crashedLastLaunch]) {
+    if (SentryClient.sharedClient && [SentryClient.sharedClient crashedLastLaunch]) {
         crashedLastLaunch = @YES;
     }
     resolve(crashedLastLaunch);
@@ -133,7 +130,6 @@ RCT_EXPORT_METHOD(startWithDsnString:(NSString * _Nonnull)dsnString options:(NSD
     NSError *error = nil;
     self.moduleMapping = [[NSMutableDictionary alloc] init];
     SentryClient *client = [[SentryClient alloc] initWithDsn:dsnString didFailWithError:&error];
-    self.client = client;
     client.beforeSerializeEvent = ^(SentryEvent * _Nonnull event) {
         [self injectReactNativeFrames:event];
         [self setReleaseVersionDist:event];

--- a/package.json
+++ b/package.json
@@ -28,8 +28,8 @@
     "react-native": ">=0.38.0"
   },
   "dependencies": {
-    "@sentry/wizard": "^0.7",
-    "raven-js": "^3.19.1"
+    "@sentry/wizard": "^0.8.1",
+    "raven-js": "^3.22.1"
   },
   "rnpm": {
     "commands": {
@@ -47,9 +47,9 @@
   },
   "devDependencies": {
     "husky": "^0.14.3",
-    "lint-staged": "^6.0.0",
-    "prettier": "^1.9.2",
-    "replace-in-file": "^3.0.0",
+    "lint-staged": "^6.0.1",
+    "prettier": "^1.10.2",
+    "replace-in-file": "^3.1.0",
     "typescript": "^2.5.3"
   },
   "lint-staged": {

